### PR TITLE
Speed up writing CustomFileEmission with buffering

### PR DESCRIPTION
**Pull request by jackkoenig**
Turns out buffering is useful.

For a medium-sized design, this reduces the time to write CHIRRTL from 60 seconds to 200-500 ms.

Adding special handling for common case of `Array[Byte]`* further reduced the time to write the file to ~60-70 ms.

*which comes from having already created a `String` in the `ChirrtlEmitter`, something we should consider doing differently

### Contributor Checklist

- [NA] Did you add Scaladoc to every public function/method?
- [NA] Did you update the FIRRTL spec to include every new feature/behavior?
- [NA] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

  - performance improvement  

#### API Impact

No impact

#### Backend Code Generation Impact

No impact

#### Desired Merge Strategy

 - Squash

#### Release Notes

Speed up writing of annotations that use `CustomFileEmission`

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (1.2.x, 1.3.0, 1.4.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

**Comment by jackkoenig** [comment]
This isn't strictly needed for `1.4.0`, I think it'd be safe to squeeze in without another release candidate though. Otherwise it can be part of `1.4.1`, although we should consider doing it relatively quickly in that case.

File: src/main/scala/firrtl/options/phases/WriteOutputAnnotations.scala [hunk 1]
```diff
@@ -37,8 +37,12 @@ class WriteOutputAnnotations extends Phase {
...
         filesWritten.get(canonical) match {
           case None =>
-            val w = new FileOutputStream(filename)
-            a.getBytes.foreach(w.write(_))
+            val w = new BufferedOutputStream(new FileOutputStream(filename))
+            a.getBytes match {
+              case arr: mutable.WrappedArray[Byte] => w.write(arr.array)
+              case other => other.foreach(w.write(_))
+            }
+            w.flush()
```
Comment (seldridge) [inline src/main/scala/firrtl/options/phases/WriteOutputAnnotations.scala:45]: Is this flush necessary given that the stream is closed on the next line?

**Review by seldridge — APPROVED** [review]
Approach looks great! Fix the 2.13 build issue and this is good to go.

File: src/main/scala/firrtl/options/phases/WriteOutputAnnotations.scala (hunk 1 above)
Comment (jackkoenig) [reply src/main/scala/firrtl/options/phases/WriteOutputAnnotations.scala:45]: I read something on StackOverflow that suggested it was, but further reading indicates it is not, will remove.

**Comment by jackkoenig** [comment]
@seldridge @chick @azidar @albert-magyar Any thoughts about merging PRs right now? This change is reasonable for 1.4.1, or snuck into an RC4 if one becomes needed (🤞 not). It's such a small change I think it could be _snuck_ into 1.4.0 without another RC but perhaps we should not go down that slippery slope.

**Comment by seldridge** [comment]
I'm in favor of just sneaking it in.

**Pull request merged**